main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java

```diff
@@ -1,15 +1,100 @@
 
 public class SmartSplitTests {
+
+    private static Properties bindings;
+    private static String FUNCTION_NAME = "smartSplit";
+
+    @BeforeSuite
+    public void setUp() {
+        bindings = new Properties();
+    }
+
+    @AfterSuite
+    public void tearDown() {
+        bindings = null;
+    }
+
+    @Test
+    public void testSmartSplitInvalidParams() {
+        assertTrue(invoke(FUNCTION_NAME) instanceof EvalError);
+        assertTrue(invoke(FUNCTION_NAME, "teststring1", 1, "teststring2", 2) instanceof EvalError);
+    }
+
+    @Test
+    public void testSmartSplitGuessComma() {
+        String testString = "teststring1,teststring2,teststring3,teststring4";
+        String[] expected = { "teststring1", "teststring2", "teststring3", "teststring4" };
+        String[] actual = (String[]) invoke(FUNCTION_NAME, testString);
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testSmartSplitGuessTab() {
+        String testString = "teststring1	teststring2	teststring3	teststring4";
+        String[] expected = { "teststring1", "teststring2", "teststring3", "teststring4" };
+        String[] actual = (String[]) invoke(FUNCTION_NAME, testString);
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testSmartSplitCharSepGiven() {
+        String testString = "teststring1#teststring2#teststring3#teststring4";
+        String[] expected = { "teststring1", "teststring2", "teststring3", "teststring4" };
+        String[] actual = (String[]) invoke(FUNCTION_NAME, testString, '#');
+        assertArrayEquals(expected, actual);
+    }
+    
+    @Test
+    public void testSmartSplitCharSepSpace() {
+        String testString = "teststring1 teststring2 teststring3 teststring4";
+        String[] expected = { "teststring1", "teststring2", "teststring3", "teststring4" };
+        String[] actual = (String[]) invoke(FUNCTION_NAME, testString, ' ');
+        assertArrayEquals(expected, actual);
+    }
+    
+    @Test
+    public void testSmartSplitStringSepGiven() {
+        String testString = "teststring1#@$teststring2#@$teststring3#@$teststring4";
+        String[] expected = { "teststring1", "teststring2", "teststring3", "teststring4" };
+        String[] actual = (String[]) invoke(FUNCTION_NAME, testString, "#@$");
+        assertArrayEquals(expected, actual);
+    }
+
+    /**
+     * Lookup a control function by name and invoke it with a variable number of
+     * args
+     */
+    private static Object invoke(String name, Object... args) {
+        // registry uses static initializer, so no need to set it up
+        Function function = ControlFunctionRegistry.getFunction(name);
+        if (function == null) {
+            throw new IllegalArgumentException("Unknown function " + name);
+        }
+        if (args == null) {
+            return function.call(bindings, new Object[0]);
+        } else {
+            return function.call(bindings, args);
+        }
+    }
```

Comment:
Fantastic, thanks a lot! As a small note, I think `invoke` should not need to take the function name as parameter, as it is always `FUNCTION_NAME` in this class (and the method is private). But that is a cosmetic concern.

Reply:
Yeah. Actually I thought we should move the invoke method to some test util class or to probably RefineTest. Actually sorry I forgot to extend SmartSplitTest class by RefineTest. I think I should probably add that code right away.

Reply:
@omkarnix moving to a test util class would make a lot of sense too.


The `RefineTest` class is used to create a test workspace and initialize various other things that are required when dealing with projects. For function tests it is not needed to extend it, as function registration is done statically as you have rightly noted.

Reply:
Ok. Thanks. I would certainly like to refactor it and move the invoke method to some util class. 
I noticed that travis CI build failed due to assertion failure in serializeProcessManager whereas appveyor build succeeded. Could it be a glitch while building on travis-ci? Can we trigger the build manually?

Reply:
> Fantastic, thanks a lot! As a small note, I think `invoke` should not need to take the function name as parameter, as it is always `FUNCTION_NAME` in this class (and the method is private). But that is a cosmetic concern.

@wetneb I agree. It was really not required to create constant for the local static value. So made the change suggested by yo .Also added one more misc change. Can you review it?

Reply:
yes don't worry about unrelated test failures (although this one is curious). I can restart the build when it happens.